File: samples/start-app/php-symfony/src/UseCase/FirstApiCall.php
<?php

declare(strict_types=1);

namespace App\UseCase;

use App\Client\ClientFactoryInterface;

class FirstApiCall
{
    public function __construct(private readonly ClientFactoryInterface $clientFactory)
    {
    }

    public function execute(): array
    {
        // Create instance of client with 'base_uri' configured in .env file and access token stored in database
        $client =  $this->clientFactory->create();

        // Replace by the API endpoint you want to call. Here an example with channels
        // https://api.akeneo.com/api-reference.html#get_channels
        $apiUrl = '/api/rest/v1/channels';

        $response =  $client->get($apiUrl);

        return json_decode($response->getBody()->getContents(), true);
    }
}

File: samples/start-app/python-fastapi/app/usecase/firstApiCall.py
from app.persistence import tokenRepository
from app.dependencies import get_config, build_user_agent
from urllib.parse import urljoin

import requests

def first_api_call_usecase(db):

    # Replace by the API endpoint you want to call. Here an example with channels
    # https://api.akeneo.com/api-reference.html#get_channels
    api_url = urljoin(get_config('AKENEO_PIM_URL'), '/api/rest/v1/channels')
    token = tokenRepository.get_token(db)

    response = requests.get(api_url,  headers={
        'Authorization': 'Bearer %s' % token.access_token,
        'User-Agent':  build_user_agent()
    })
    
    return response
